src/HftApi/WebApi/OperationsController.cs

```diff
@@ -0,0 +1,285 @@
+                    FeeSettings = new FeeSettingsCashoutModel
+                    {
+                        TargetClients = new Dictionary<string, string>
+                        {
+                            { "Cashout", _feeSettings.WithdrawalFeeDestinationClientId }
+                        }
+                    }
+                }
+            };
+
+            _cqrsEngine.SendCommand(cashoutCommand, "hft-api", OperationsBoundedContext.Name);
+
+            return Ok(ResponseModel<Guid>.Ok(operationId));
+        }
+        
+        /// <summary>
+        /// Get Withdrawal by Id
+        /// </summary>
+        /// <param name="withdrawalId"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("withdrawals/{withdrawalId}")]
+        [ProducesResponseType(typeof(ResponseModel<WithdrawalModel>), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> Get(Guid withdrawalId)
+        {
+            OperationModel operation = null;
+
+            try
+            {
+                operation = await _operationsClient.Operations.Get(withdrawalId);
+                
+                JObject.Parse(operation.ContextJson).TryGetValue("WalletId", out var walletId);
+                
+                if(walletId == null || operation.Type != OperationType.Cashout || walletId.Value<string>() != User.GetWalletId())
+                    throw HftApiException.Create(HftApiErrorCode.ItemNotFound, "Withdrawal not found");
+            }
+            catch (HttpOperationException e)
+            {
+                if (e.Response.StatusCode == HttpStatusCode.NotFound)
+                    throw HftApiException.Create(HftApiErrorCode.ItemNotFound, "Withdrawal not found");
+
+                throw;
+            }
+
+            if (operation == null)
+                throw HftApiException.Create(HftApiErrorCode.ItemNotFound, "Withdrawal not found");
+
+            return Ok(ResponseModel<WithdrawalModel>.Ok(_mapper.Map<WithdrawalModel>(operation)));
+        }
+
+        private async Task<Asset> CheckDepositPreconditionsAsync(string assetId=default)
+        {
+            Asset asset = default;
+            
+            if (!string.IsNullOrWhiteSpace(assetId))
+            {
+                asset = await _assetsService.GetAssetByIdAsync(assetId);
+
+                if (asset == null || asset.IsDisabled || !asset.BlockchainDepositEnabled)
+                    throw HftApiException.Create(HftApiErrorCode.ItemNotFound, "Asset not found");
+
+                var assetsAvailableToClient = await _assetsService.GetAllAssetsAsync(User.GetClientId());
+
+                if (assetsAvailableToClient.SingleOrDefault(x => x.AssetId == assetId) == null)
+                    throw HftApiException.Create(HftApiErrorCode.ActionForbidden, "Asset unavailable");
+            }
+
+            var pendingDialogs = await _clientDialogsClient.ClientDialogs.GetDialogsAsync(User.GetClientId());
```

Comment:
do we need to check for pending dialogs for HFT wallets?

Reply:
Yes, confirmed with Ignacio